### PR TITLE
Code cleanup for hub device common folder

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Common/ExceptionTrace.cs [hunk 1]
```diff
@@ -103,50 +103,19 @@ public TException TraceException<TException>(TException exception, TraceEventTyp
...
 
+#if NET451
                 switch (level)
                 {
                     case TraceEventType.Critical:
                     case TraceEventType.Error:
-#if NET451
-                        Trace.TraceError("An Exception is being thrown: {0}", GetDetailsForThrownException(exception));
-#endif
-                        ////if (MessagingClientEtwProvider.Provider.IsEnabled(
```
Comment (drwill-ms) [inline iothub/device/src/Common/ExceptionTrace.cs:113]: I presume this is dead code, so I just removed it. @timtay-microsoft, any context to sway this change?

Comment (timtay-microsoft) [reply iothub/device/src/Common/ExceptionTrace.cs:113]: No objections here. This class as a whole seems out of place so I'm not sure what the commented out code was for.

File: iothub/device/src/Common/Exceptions/IotHubException.cs [hunk 2]
```diff
@@ -19,14 +19,14 @@ public class IotHubException : Exception
         private const string TrackingIdValueSerializationStoreName = "IotHubException-TrackingId";
 
         /// <summary>
-        /// Creates an instance of <see cref="IotHubException"/> an empty error message.
+        /// Creates an instance of this class an empty error message.
```
Comment (timtay-microsoft) [inline iothub/device/src/Common/Exceptions/IotHubException.cs:22]:
```suggestion
        /// Creates an instance of this class with an empty error message.
```

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**